sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ContainerThroughputControlGroupProperties.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.throughputControl;
+
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.throughputControl.config.ThroughputControlGroupInternal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public class ContainerThroughputControlGroupProperties {
+    private static Logger logger  = LoggerFactory.getLogger(ContainerThroughputControlGroupProperties.class);
+
+    private final AtomicReference<ThroughputControlGroupInternal> defaultGroup;
+    private final Set<ThroughputControlGroupInternal> throughputControlGroupSet;
+    private final Set<String> supressInitErrorGroupSet;
+
+
+    public ContainerThroughputControlGroupProperties() {
+        this.defaultGroup = new AtomicReference<>();
+        this.throughputControlGroupSet = ConcurrentHashMap.newKeySet();
+        this.supressInitErrorGroupSet = ConcurrentHashMap.newKeySet();
+    }
+
+    public int addThroughputControlGroup(ThroughputControlGroupInternal group) {
+        checkNotNull(group, "Throughput control group should not be null");
+
+        if (group.isDefault()) {
+            if (!this.defaultGroup.compareAndSet(null, group)) {
+                if (!this.defaultGroup.get().equals(group)) {
+                    throw new IllegalArgumentException("A default group already exists");
+                }
+            }
+        }
+
+        if (group.isContinueOnInitError()) {
+            this.supressInitErrorGroupSet.add(group.getGroupName());
+        }
+
+        // Only throw when two different groups are using the same id (databaseId + containerId + groupName)
+        if (this.throughputControlGroupSet.stream()
+                .anyMatch(existingGroup -> StringUtils.equals(existingGroup.getId(), group.getId()) && !existingGroup.equals(group))) {
```

Comment:
Instead of using StringUtils from apache, we should use `Objects.equals(id, that.id)` - which is coming from java.util package - which is also the default implementation of equals method.

Reply:
will update in next iteration.